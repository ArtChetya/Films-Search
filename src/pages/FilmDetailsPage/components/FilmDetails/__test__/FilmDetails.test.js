import { shallow } from 'enzyme/build';
import React from 'react';
import { FilmDetails } from '..';

describe('FilmDetails component', () => {
    let filmDetails;

    beforeAll(() => {
        filmDetails = {
            id: 68718,
            title: 'Django Unchained',
            tagline: 'Life, liberty and the pursuit of vengeance.',
            vote_average: 7.9,
            vote_count: 12144,
            release_date: '2012-12-25',
            poster_path:
                'https://image.tmdb.org/t/p/w500/5WJnxuw41sddupf8cwOxYftuvJG.jpg',
            overview:
                'With the help of a German bounty hunter, a freed slave sets out to rescue his wife from a brutal Mississippi plantation owner.',
            budget: 100000000,
            revenue: 425368238,
            genres: ['Drama', 'Western'],
            runtime: 165
        };
    });

    it('should be rendered correctly', () => {
        const component = shallow(<FilmDetails filmDetails={filmDetails} />);
        expect(component).toMatchSnapshot();
    });
});
