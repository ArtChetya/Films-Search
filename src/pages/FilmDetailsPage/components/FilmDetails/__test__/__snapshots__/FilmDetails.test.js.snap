// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetails component should be rendered correctly 1`] = `
<WithStyles(Grid)
  container={true}
  wrap="nowrap"
>
  <WithStyles(Grid)
    item={true}
    xs={3}
  >
    <FilmDetails__Poster
      alt="Film Poster"
      src="https://image.tmdb.org/t/p/w500/5WJnxuw41sddupf8cwOxYftuvJG.jpg"
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    direction="column"
    item={true}
    spacing={8}
  >
    <WithStyles(Grid)
      container={true}
      item={true}
    >
      <ColorText
        fontcolor="#ec407a"
        variant="h5"
      >
         
        Django Unchained
         
      </ColorText>
      <FilmDetails__Rating
        label={7.9}
        variant="outlined"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    >
      <ColorText
        fontcolor="#e0e0e0"
        varaint="body1"
      >
        Life, liberty and the pursuit of vengeance.
      </ColorText>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      item={true}
    >
      <ColorText
        varaint="subtitle2"
      >
        2012
      </ColorText>
      <FilmDetails__MarginedText
        varaint="subtitle2"
      >
        165
         min
      </FilmDetails__MarginedText>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    >
      <ColorText
        fontcolor="#e0e0e0"
        varaint="body1"
      >
        With the help of a German bounty hunter, a freed slave sets out to rescue his wife from a brutal Mississippi plantation owner.
      </ColorText>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
